opentelemetry/proto/profiles/v1development/eventprofile.md

```diff
@@ -0,0 +1,284 @@
+        attributes:  [1]     # thread_id: 2
+    },
+    {
+        event_type_index: 0    # cpu_sample (can be omitted on the wire)
+        stack_index: 1         # main;serve_request
+        time:        9_200_000 # 9.2ms since previous event
+        attributes:  [0]       # thread_id: 1
+    }
+]
+# stacks, locations, functions, mappings, etc. omitted for brevity
+```
+
+### Timestamped CPU Profile + Off-CPU Thread State Profile
+
+
+```txtpb
+event_types: [
+    { # index 0
+        name_string_index: 1  # "cpu_sample"
+        unit_string_index: 2  # "ms"
+        default_value:     10 # each event represents 10ms of CPU time
+        clock_id:          0
+    } 
+    { # index 1
+        name_string_index: 2  # "thread_state"
+        clock_id:          0
+    }
+]
+clocks: [
+    { # index 0
+        frequency :      1_000_000_000          # 1 Ghz clock, i.e. 1 cycle = 1 ns
+        time_unix_nanos: 1_257_894_000_000_000_000
+    }
+]
+attributes: [
+    { # index 0
+        key_index: 4          # thread.id
+        value: {int_value: 1} # 1
+    },
+    { # index 1
+        key_index: 4,         # thread.id
+        value: {int_value: 2} # 2
+    }
+    { # index 2
+        key_index: 5,         # thread.state
+        value: {key_index: 6} # mutex
+    }, # index 3
+    {
+        key_index: 5,         # thread.state
+        value: {key_index: 7} # sleep
+    }
+]
+strings: [
+    "",             # index 0
+    "cpu_sample",   # index 1
+    "thread_state", # index 2
+    "ms",           # index 3
+    "thread.id",    # index 4
+    "thread.state", # index 5
+    "mutex",        # index 6
+    "sleep",        # index 7
+]
+
+events: [
+    {
+        event_type_index: 0     # cpu_sample (can be omitted on the wire)
+        stack_index: 1          # main;serve_request
+        time:        10_500_000 # 10.5ms since clock start
+        attributes:  [0]        # thread_id: 1
+    }
+    {
+        event_type_index: 0     # cpu_sample (can be omitted on the wire)
+        stack_index: 2          # main;foo
+        time:        300_000    # 0.3ms since previous event
+        attributes: [1]         # thread_id: 2
+    },
+    {
+        event_type_index: 0     # cpu_sample (can be omitted on the wire)
+        stack_index: 1          # main;serve_request
+        time:        9_200_000  # 9.2ms since previous event
+        attributes:  [0]        # thread_id: 1
+    },
+    {
+        event_type_index: 1            # thread_state
+        stack_index:      1            # main;serve_request;mutex_lock
+        time:             300_000_000  # 300ms since previous event
+        duration:         50_000_500   # 50ms
+        attributes:       [0, 2]       # thread_id: 1, thread.state: mutex
+    },
+    {
+        event_type_index: 1             # thread_state
+        stack_index:      1             # main;foo;sleep
+        time:             5_000_000     # 5ms since previous event
```

Comment:
IMHO, if we had a duration in addition to time, we could make off_cpu and stall events clearer. IE: Stall started at time X, lasted for Y. This gives a start/stop. This could be stored in an attribute, but feels somewhat wasteful to have to have an attribute per-duration value.

Reply:
duration and value are separate fields in the current proposal, so I think the use cases brought up here should be covered? https://github.com/open-telemetry/opentelemetry-proto/pull/594/files/7d85bba21098f7950813d9717f06e86865da0726#diff-4a0777f946b5d445b17b64906a2ee5b121a070ee499189a656988ad0df350e11R81-R94

Reply:
Yeah, sorry I missed that! Definitely covers what I was asking for, thanks!